Index: src/c/src/zookeeper.c
===================================================================
--- src/c/src/zookeeper.c	(revision 1530160)
+++ src/c/src/zookeeper.c	(working copy)
@@ -1650,13 +1650,15 @@
         // a PING
         if (zh->state==ZOO_CONNECTED_STATE) {
             send_to = zh->recv_timeout/3 - idle_send;
-            if (send_to <= 0 && zh->sent_requests.head==0) {
-//                LOG_DEBUG(("Sending PING to %s (exceeded idle by %dms)",
-//                                format_current_endpoint_info(zh),-send_to));
-                int rc=send_ping(zh);
-                if (rc < 0){
-                    LOG_ERROR(("failed to send PING request (zk retcode=%d)",rc));
-                    return api_epilog(zh,rc);
+            if (send_to <= 0) {
+                if (zh->sent_requests.head==0) {
+//                    LOG_DEBUG(("Sending PING to %s (exceeded idle by %dms)",
+//                                    format_current_endpoint_info(zh),-send_to));
+                    int rc=send_ping(zh);
+                    if (rc < 0){
+                        LOG_ERROR(("failed to send PING request (zk retcode=%d)",rc));
+                        return api_epilog(zh,rc);
+                    }
                 }
                 send_to = zh->recv_timeout/3;
             }
